### PR TITLE
unison-local-ui 0.5.16

### PR DESCRIPTION
**Pull request by aryairani**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by aryairani** [comment]
Oop, it looks like CI is blocked on the number of stars on the repo. It's not really a standalone formula, it's static resources built for https://github.com/unisonweb/unison/ (which has more stars), but putting it into a separate formula so that it can be shared by the separate builds for unison, for which I'd update the formula separately.

Or should I put both formula into the same commit?

I created this PR to debug bottle creation of the resource formula, per @SMillerDev's advice.

Thanks/sorry.

**Comment by p-linnane** [comment]
I've added a label that will allow the builds to continue running despite one failing. That should hopefully get you what you need.

**Comment by aryairani** [comment]
Thank you for the no-fail-fast label — 

Gosh though, it appears to be failing on a test that simply checks that the correct files were installed during install, and I don't see any errors or clues as to why they would not have been installed.

Sorry to be a bother, but any suggestions as to how I can debug this? It works fine for me on a fresh account on my machine.

**Comment by aryairani** [comment]
Well, I'll try running `find` to see what is getting installed. The arm64 installs are failing for a different reason than the x86_64 installs; again weird because I tested on arm64 locally.

**Comment by p-linnane** [comment]
It looks like this actually gets installed in our main `unisonlang` formula. Please take a look at let me know if I'm misunderstanding here since I do not know anything about this software.

https://github.com/Homebrew/homebrew-core/blob/179c750b6730fb803c70c3afc77553551f4973da/Formula/u/unisonlang.rb#L44-L48

https://github.com/Homebrew/homebrew-core/blob/179c750b6730fb803c70c3afc77553551f4973da/Formula/u/unisonlang.rb#L54-L62

**Comment by aryairani** [comment]
Hi @p-linnane, 

Thanks — you're not misunderstanding.  `unisonlang.rb` does currently build both `unisonweb/unison` and `unisonweb/unison-local-ui`, but I was going to propose splitting them up. 

Currently it `unison-local-ui` is getting built for each platform that `unisonweb/unison` is built for, even though it's just a static asset that doesn't differ between platforms. Also, the two repos don't necessarily receive updates at the same time; `unisonweb/unison` gets updated much more frequently.

I'm one of the maintainers of `unisonweb/unison` and `unisonweb/unison-local-ui`, and I was preparing with this PR to update `unisonlang.rb` as well, to _not_ build `unison-local-ui`, but instead depend on this `unison-local-ui` package.  Granted, I am not clear on whether splitting the static assets into their own package to avoid being built 7x+ into 7 different bottles instead of 1x _is_ a better, homebrewy way to do things, and advice is welcome! (I had started a discussion about it [here](https://github.com/orgs/Homebrew/discussions/5105#discussioncomment-8388571) which led me to creating this PR.)

I was wanting to go further with this PR and debug any bottling surprises that come up, but it looks like I'm stuck for the moment due to 
> * GitHub repository not notable enough (<30 forks, <30 watchers and <75 stars)

on `unisonweb/unison-local-ui`. Is that something we can bypass for testing purposes with a tag as well?

In summary, if you know the answers to any or all of these:
Q1: Is it a good homebrewy idea to split the assets off to avoid lots of unnecessary building and versioning uncertainty? I'm not sure personally, but was leaning "yes". (I want to do it for the brew tap we maintain at least, to cut build times.)

Q2: If so, can we bypass "repo not notable enough"?

Q3: Should I file a bug report on the `node.rb` arm64 build, which is apparently installing x86_64 binaries for the elm module? (See your second source snippet above vs [`e5dca2e` (#161979)](https://github.com/Homebrew/homebrew-core/pull/161979/commits/e5dca2ec4764f8c7169a0bf2407c38947ddd9f6b), both questionable workarounds.)

Thank you!

**Comment by p-linnane** [comment]
> In summary, if you know the answers to any or all of these: Q1: Is it a good homebrewy idea to split the assets off to avoid lots of unnecessary building and versioning uncertainty? I'm not sure personally, but was leaning "yes". (I want to do it for the brew tap we maintain at least, to cut build times.)
> 
> Q2: If so, can we bypass "repo not notable enough"?
> 
> Q3: Should I file a bug report on the `node.rb` arm64 build, which is apparently installing x86_64 binaries for the elm module? (See your second source snippet above vs [`e5dca2e` (#161979)](https://github.com/Homebrew/homebrew-core/pull/161979/commits/e5dca2ec4764f8c7169a0bf2407c38947ddd9f6b), both questionable workarounds.)

Thanks for the detailed write up here. Very helpful for me.

Q1: I would lean towards not packaging something on it's own if it is not useful without another component. There is not a clear rule that I am aware of, so I will ask the rest of @Homebrew/maintainers to take a look here and chime in.

Q2: We use that check to make sure we are not adding obscure and often self-submitted software. We can override it if there is a good reason though. That check would not affect any part of the build or test though, so if there are other issues they are unrelated to this.

Q3: Opening a PR if you have a good idea of how to improve the workarounds in `node` would be appreciated and we're happy to take a look. We cannot require or install Rosetta, so as far as I know this is currently our best option.

**Comment by ZhongRuoyu** [comment]
I agree that making a separate formula is not a good idea if it shouldn't/can't be used standalone. My comments on "building and versioning uncertainty" specifically:

> splitting the static assets into their own package to avoid being built 7x+ into 7 different bottles instead of 1x

Even if we package `unison-local-ui` separately, we would still need to build the 7 bottles, actually.

> Also, the two repos don't necessarily receive updates at the same time; `unisonweb/unison` gets updated much more frequently.

I understand the concern regarding saving build times (and I appreciate that), but given the moderate frequency of `unisonlang` updates (~12 in the past year), and that `unison-local-ui` itself (this PR) only takes <8min to finish, I think it's okay to keep that as a resource.

**Comment by aryairani** [comment]
Thanks for the feedback @ZhongRuoyu!

> Even if we package `unison-local-ui` separately, we would still need to build the 7 bottles, actually.

Could you explain a bit more about this?  I had gotten a different impression from  https://github.com/orgs/Homebrew/discussions/5105#discussioncomment-8372365, specifically I interpreted it as that it was possible, and thus desirable, to perform a build that resulted in a single, multi-platform bottle.

Admittedly, I couldn't find anything in the docs about multi-platform bottles, and if they don't exist then my approach with this PR seems pointless 😅    So, do they not exist?  (And of course, like you said, we can stick with the old approach regardless, but I am still curious.)

I will also look into creating a pre-built resource tarball on our end that doesn't require a build step in homebrew.
(And that would be fine, right? There's no practice of requiring that every piece of the formula should be built from scratch within the formula is there? Or _is_ there?) :)

**Comment by ZhongRuoyu** [comment]
> Could you explain a bit more about this? I had gotten a different impression from https://github.com/orgs/Homebrew/discussions/5105#discussioncomment-8372365, specifically I interpreted it as that it was possible, and thus desirable, to perform a build that resulted in a single, multi-platform bottle.
>
> Admittedly, I couldn't find anything in the docs about multi-platform bottles, and if they don't exist then my approach with this PR seems pointless 😅 So, do they not exist? (And of course, like you said, we can stick with the old approach regardless, but I am still curious.)

https://github.com/orgs/Homebrew/discussions/5105#discussioncomment-8372365 means if the built bottles are identical across all platforms, an `all` bottle will be produced. Take `google-java-format` as an example:

https://github.com/Homebrew/homebrew-core/blob/a7bfe3537c89b76c1f07cd09df201c8ba2b44ff8/Formula/g/google-java-format.rb#L10-L12

The bottles contain a cross-platform jar (Java Archive) and a Python script, which turned out to be identical on all platforms, so an `all` bottle was produced. But currently producing an `all` bottle *does* require the formula to be built on all platforms, hence the "7 bottles".

> I will also look into creating a pre-built resource tarball on our end that doesn't require a build step in homebrew. (And that would be fine, right? There's no practice of requiring that every piece of the formula should be built from scratch within the formula is there? Or _is_ there?) :)

[We don't like prebuilt binaries](https://docs.brew.sh/Acceptable-Formulae#we-dont-like-binary-formulae), sorry.

**Comment by MikeMcQuaid** [comment]
> Q1: I would lean towards not packaging something on it's own if it is not useful without another component. There is not a clear rule that I am aware of, so I will ask the rest of @Homebrew/maintainers to take a look here and chime in.

This is a rule but it's perhaps not written down anywhere explicitly (but should be).

**Comment by aryairani** [comment]
> The bottles contain a cross-platform jar (Java Archive) and a Python script, which turned out to be identical on all platforms, so an `all` bottle was produced. But currently producing an `all` bottle _does_ require the formula to be built on all platforms, hence the "7 bottles".

Ah! Ok, I understand.

> > I will also look into creating a pre-built resource tarball on our end that doesn't require a build step in homebrew. (And that would be fine, right? There's no practice of requiring that every piece of the formula should be built from scratch within the formula is there? Or _is_ there?) :)
> 
> [We don't like prebuilt binaries](https://docs.brew.sh/Acceptable-Formulae#we-dont-like-binary-formulae), sorry.

Just to clarify, this would not be binaries, it's static html/css resouces that got built with node (by the formula in this PR).
